contrib/metrics/supply_check/supply_deltas.py

```diff
@@ -0,0 +1,73 @@
+import pprint
+
+class SupplyDeltas:
+    def __init__(self, zcashd, fr_addrs, miner_deltas, flush_interval = 500):
+        self.zcashd = zcashd
+        self.fr_addrs = fr_addrs
+        self.miner_deltas = miner_deltas
+
+        self.delta_count = 0
+        self.delta_total = 0
+        self.delta_cache = []
+        self.flush_interval = flush_interval
+        
+        deltas_flat = [pair for deltas in miner_deltas.values() for pair in deltas]
+        for (deltaHeight, delta) in sorted(deltas_flat):
+            self.AddSupplyDelta(deltaHeight, delta)
+
+    def AddSupplyDelta(self, deltaHeight, delta):
+        assert len(self.delta_cache) == 0 or deltaHeight > self.delta_cache[-1][0]
+
+        self.delta_count += 1
+        self.delta_total += delta
+        self.delta_cache.append((deltaHeight, self.delta_total))
+        
+    def DeviationAtHeight(self, height):
+        # search for the cached entry at the maximum deltaHeight <= height
+        def binary_search(start, end, max_found):
```

Comment:
Binary search is in the standard library: https://docs.python.org/3/library/bisect.html